### PR TITLE
zephyr: misc fixes for build on zephyr

### PR DESCRIPTION
**Pull request by lgirdwood**
Numerous small fixes that fix build issues on Zephyr.

### CODE REVIEW

File: src/include/sof/lib/alloc.h [hunk 1]
```diff
@@ -24,6 +24,9 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <sof/lib/cache.h>
+#include <sof/lib/memory.h>
+
```
Comment (paulstelian97) [inline src/include/sof/lib/alloc.h:29]: Won't approve until you have actually checked that these are indeed needed and updated commit message to no longer state "TODO check."

Comment (lgirdwood) [reply src/include/sof/lib/alloc.h:29]: you are right - this one has not been checked.

Comment (lgirdwood) [reply src/include/sof/lib/alloc.h:29]: ......and it's not needed now. Will remove.

**Comment by lgirdwood** [comment]
CI unrelated module load stress.

**Pull request merged**